src/Controllers/QuizController.php

```diff
@@ -92,4 +92,90 @@ public function read($id)
             return ApiResponse::failed($e);
         }
     }
+
+    public function edit(Request $request, $id)
+    {
+        try {
+            $user = auth()->user();
+            $request->validate([
+                'title' => 'string|max:255',
+                'description' => 'nullable|string|max:65535',
+                'start_time' => 'nullable|date_format:Y-m-d\TH:i:sp',
+                'end_time' => 'nullable|date_format:Y-m-d\TH:i:sp',
+                'duration' => 'nullable|integer',
+                'point' => 'nullable|integer',
+                'link_url' => 'nullable|string|max:65535',
+            ]);
+
+            $quiz = Quiz::find($id);
+            if ($quiz?->created_by != $user->id) {
+                throw ValidationException::withMessages([
+                    'id' => 'Quiz not found',
+                ]);
+            }
+
+            if (! CourseUserHelper::isUserInCourse(
+                $user->id,
+                $quiz->course_id,
+                CourseUserRole::TEACHER,
+            )) {
+                throw ValidationException::withMessages([
+                    'id' => 'Must enroll the course to edit the quiz',
+                ]);
+            }
+
+            $quiz->fill($request->only([
+                'title',
+                'description',
+                'start_time',
+                'end_time',
+                'duration',
+                'point',
+                'link_url',
+            ]) + [
+                'start_time' => gmdate(
+                    'Y-m-d H:i:s',
+                    strtotime($request->input('start_time')),
+                ),
+                'end_time' => gmdate(
+                    'Y-m-d H:i:s',
+                    strtotime($request->input('end_time')),
+                ),
```

Comment:
not really sure about this one, but as far as I remember, putting `'start_time'` and `'end_time'` in both arrays will make your second array useless because the value from the first array won't be overridden. *CMIIW

example: https://www.tehplayground.com/OmNMko9toURos8o3

ref: https://github.com/uasoft-indonesia/badaso-lms-module/blob/41bb1e4c72c8402cc99a2bade558fa5d31cd2dd4/src/Controllers/AssignmentController.php#L130-L142